keps/sig-node/2008-forensic-container-checkpointing/README.md

```diff
@@ -292,14 +419,111 @@ Checkpointing containers will be possible again.
+If the underlying container engine does not support or if it is not implemented,
+the kubelet API endpoint will fail with `500`. The error code is the same if the
+container engine does not implement it explicitly or if the underlying container
+engine is too old. The difference between does two failures are only visible in
+the error message returned from the container engine.
+
+If the underlying container engine does not support the CRI RPC the kubelet API endpoint
+will always return `500`.
+
+It cannot directly impact running workloads, but if the *kubelet* API endpoint is
+called if the underlying container engine does no longer support it, the checkpoint
+request will fail.
+
+###### What specific metrics should inform a rollback?
+
+CRI metrics will be added to track checkpointing failures to inform a rollback
+decision.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No data is stored, so re-enabling starts from a clean slate.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+Querying the state of the feature gate offers the possibility to detect
+if the API endpoint will return `404` or not.
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+As it is not exposed in the Kubernetes API it cannot be determined. This is
+only visible in the kubelet. Also, this is a feature workloads are not using
+directly, but which is only external entities can trigger. Access to the
+*kubelet* API endpoints is needed. It might be detectable if operators can
+query the state of different feature gates. An operator could also use metrics
+to determine that this feature is in use.
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+###### How can someone using this feature know that it is working for their instance?
+
+The kubelet API endpoint can return following codes:
+
+- 200: checkpoint archive was successfully created
+- 404: feature is not enabled
+- 500: underlying container engine does not support checkpointing containers
+
+Documented at <https://kubernetes.io/docs/reference/node/kubelet-checkpoint-api/>
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Does not apply as the enhancement will only be called when requested. Not a service.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+Does not apply as the enhancement will only be called when requested. Not a service.
```

Comment:
These two need updates to indicate that the kubelet will add (or does have) metrics for the kubelet endpoints indicating usage counts and failure counts.  Prior to going to beta, the exact metric names must be added.

Reply:
Reworded as suggested